pkg/internal/cmd/cmd.go

```diff
@@ -143,3 +141,19 @@ func NewCommand(ctx context.Context) *cobra.Command {
 
 	return cmd
 }
+
+type dynamicCertRunnable struct {
```

Comment:
Could we make this something generic? So we can re-use it in cert-manager (I think I made the same mistake there).

Reply:
I am not sure if that's possible, but I am not familiar with Golang generics. But I know that generics is not supported for methods, only functions.

Reply:
But feel free to open another PR to fix this. My main concern is to fix the bug.

Reply:
I meant making it generic by letting it accept a function as argument (not using actual generics).

Reply:
Ahhh, sorry for being a bit slow. Can we fix this by making the `DynamicSource` implement these interfaces? That will make it easier to use across projects.